test/functional/fixtures/reporter/test.js

```diff
@@ -1133,6 +1133,142 @@ const experimentalDebug = !!process.env.EXPERIMENTAL_DEBUG;
 
     });
 
+    describe('Reporter Hooks', () => {
+        it('Should handle onBeforeWrite hook', () => {
+            function custom () {
+                return {
+                    taskInfo: { prop: 'val' },
+                    reportTaskStart (...args) {
+                        this.setIndent(3).write('unFormattedText', args);
```

Comment:
One more clarification: one method of reporter plugin host (for exampel `reportTaskStart`), one custom reporter method and one method without name (anonymous function).